File: 20-memory/npm-package/memory.ts
import axios from 'axios';
const API = 'https://cost-katana-backend.store/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function storeMemory(agentId: string, key: string, value: any) {
  const res = await axios.post(`${API}/memory`, {agentId, key, value},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Memory stored');
  return res.data.data;
}

export async function getMemory(agentId: string, key: string) {
  const res = await axios.get(`${API}/memory/${agentId}/${key}`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('📖 Memory:', res.data.data);
  return res.data.data;
}

File: 20-memory/python-sdk/memory.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def store_memory(agent_id, key, value):
    res = requests.post(f"{API}/memory",
        json={"agentId": agent_id, "key": key, "value": value},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print("✅ Memory stored")
    return res.json()["data"]

def get_memory(agent_id, key):
    res = requests.get(f"{API}/memory/{agent_id}/{key}",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"📖 Memory: {res.json()['data']}")
    return res.json()["data"]
